samples/apps/copilot-chat-app/webapp/src/redux/features/conversations/conversationsSlice.ts

```diff
@@ -58,17 +77,34 @@ export const conversationsSlice: Slice<ConversationsState> = createSlice({
             state.conversations[id].messages[messageIndex].state = newMessageState;
             frontLoadChat(state, id);
         },
+        updateUserIsTyping: (state: ConversationsState, action: PayloadAction<{ userId: string; chatId: string; isTyping: boolean }>) => {
+            const { userId, chatId, isTyping } = action.payload;
+            updateUserTypingState(state, userId, chatId, isTyping);
+        },
+        updateUserIsTypingFromServer: (state: ConversationsState, action: PayloadAction<{ userId: string; chatId: string; isTyping: boolean }>) => {
+            const { userId, chatId, isTyping } = action.payload;
+            updateUserTypingState(state, userId, chatId, isTyping);
+        },
```

Comment:
do we need both of these functions? I understand that they are used in separate places, but they have the same signature and do the same thing...

Reply:
Yes! This is so we can have different function handles for signalR callbacks; e.g updating state when its generated only from the backend (so broadcast to everyone) or when state is updated by a user (so broadcast to everyone but that user).

Reply:
They are called by different actors. 
`updateUserIsTyping` is called by the webapp itself when the user starts typing. And this function call will be captured by the redux middleware to broadcast the state.
`updateUserIsTypingFromServer` is called when the app receives the state broadcasted by the server.

If you use only one function, we will create an infinite loop because `updateUsersTyping` will trigger a broadcast event.

Reply:
Understood now, but can you add a note in the README to explain this better? Was also super confusing to me until this last message

Reply:
Done